packages/babel-traverse/src/path/modification.ts

```diff
@@ -127,6 +130,18 @@ export function insertAfter(this: NodePath, nodes_: t.Node | t.Node[]) {
     if (this.node) {
       const node = this.node as t.Expression | t.VariableDeclaration;
       let { scope } = this;
+
+      if (scope.path.isPattern()) {
+        this.replaceWith(
+          t.callExpression(
+            t.arrowFunctionExpression([], this.node as t.Expression),
```

Comment:
nit: use invariant instead if node will never be `t.VariableDeclaration` when `scope.path` is pattern.

Reply:
Oh I just realize that it is TypeScript! so we can use https://github.com/microsoft/TypeScript/pull/32695 instead.